sklearn/utils/__init__.py

```diff
@@ -304,10 +247,19 @@ def _check_key_type(key, superclass):
     if isinstance(key, superclass):
         return True
     if isinstance(key, slice):
+        if key.start is None and key.stop is None:
+            return False
         return (isinstance(key.start, (superclass, type(None))) and
                 isinstance(key.stop, (superclass, type(None))))
     if isinstance(key, list):
-        return all(isinstance(x, superclass) for x in key)
+        unique_key = set(key)
+        all_superclass = all(isinstance(x, superclass) for x in unique_key)
+        if superclass != int:
+            return all_superclass
+        # bool is a subclass of int, therefore we should check specifically for
+        # any bool
+        any_bool = any(isinstance(x, bool) for x in unique_key)
```

Comment:
should this also consider `np.bool_` or do we assume it's unnecessary because `key` is passed a standard python type?

Reply:
I added this support